### PR TITLE
[new] scripts/update-schema-latest.sh

### PR DESCRIPTION
**Pull request by TakahikoKawasaki**
This script does the following.

1. Switch to the 'gh-pages' branch.

2. Copy "schemas/v{major}.{minor}/schema.json" in the 'main' branch to "oas/{major}.{minor}/schema/{YYYY-MM-DD}" in the 'gh-pages' branch. The value of {YYYY-MM-DD} is extracted from the '$id' property in the schema.json file.

3. Create a symbolic link "latest" under "oas/{major}.{minor}/schema" that points to the "{YYYY-MM-DD}" file.

4. Copy "schemas/v{major}.{minor}/schema-base.json" in the 'main' branch to "oas/{major}.{minor}/schema-base/{YYYY-MM-DD}" in the 'gh-pages' branch. The value of {YYYY-MM-DD} is extracted from the '$id' property in the schema-base.json file.

5. Create a symbolic link "latest" under "oas/{major}.{minor}/schema-base" that points to the "{YYYY-MM-DD}" file.

### CODE REVIEW

**Review by ralfhandl — CHANGES_REQUESTED** [review]
Wrong PR 🙄

**Comment by ralfhandl** [comment]
Hi @TakahikoKawasaki,

Thanks for providing the script. 

The script works fine for the v3.1 case with separate `schema` and `schema-base`, and fail for the v3.0 case where there is only `schema`.

Would you mind adding this distinction to the scripts?

Thanks in advance!

**Comment by TakahikoKawasaki** [comment]
@ralfhandl
- Updated the script to support the version 3.0.
- Versions older than 3.0 are rejected on the assumption that maintenance of 2.x and older versions has stopped.

If it is necessary to support versions older than 3.0, please let me know.

File: scripts/update-schema-latest.sh [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+    # The command to copy the source file to the destination file.
+    copy_command="git show main:${src_file} > ${dst_file}"
+
+    # The command to update the 'latest' file.
+    link_command="cd ${dst_dir} && ln -sf ${date} latest"
+
+    # Execute the commands.
+    __execute_command "${copy_command}"
+    __execute_command "${link_command}"
```
Comment (ralfhandl) [inline scripts/update-schema-latest.sh:346]: If #4034 is approved, this needs to become

```suggestion
    # The commands to update the 'latest' file.
    link_command="cd ${dst_dir} && ln -sf ${date} latest"
    link_command2="cd ${dst_dir} && ln -sf ${date} latest.json"

    # Execute the commands.
    __execute_command "${copy_command}"
    __execute_command "${link_command}"
    __execute_command "${link_command2}"
```

**Review by limck5856 — APPROVED** [review]

**Comment by ralfhandl** [comment]
Steps 3 and 5 (symlink `latest`) need to be replaced with
* rename `{YYYY-MM-DD}.md` to match the date of the new schema file

**Review by ralfhandl — CHANGES_REQUESTED** [review]
Steps 3 and 5 (symlink `latest`) need to be replaced with
* rename `{YYYY-MM-DD}.md` to match the date of the new schema file

**Comment by lornajane** [comment]
I'm closing this pull request since it's no longer needed since the recent changes to the schema publishing process.

**Pull request closed without merging**